### ISSUE TITLE
Error: Authentication failed. Make sure that the AWS profile is set correctly.

### ISSUE THREAD

**Issue by PodStuart** [milliHQ/terraform-aws-next-js#333]
I've followed the steps in the atomic-deployments example. The only changes I've made are to main.tf to add the version of the tf_next module (1.0.0-canary.4) and to specify my aws region/profile. The terraform steps have all run successfully and I am not trying to run the tf-next deploy and getting the following error.

```
Error: Authentication failed. Make sure that the AWS profile is set correctly.
Could not upload deployment package.
```

I've tried adding --awsProfile but this results in an unknown argument error. I tried --profile and that got the same authentication error above. I'm a bit confused about the correct way to specify the profile and I think that the documentation could be out of sync with the CLI itself.

In the end we found that setting AWS_PROFILE as an environment variable got the deployment to work. Perhaps the documentation could be updated to make this a little easier to understand?

**Comment by PodStuart** [comment]
Thanks very much for the swift and detailed response. We've got things up and running now. Really cool to see how easy it is to deploy new webapp versions and flip between them using aliases.

**Comment by soerface** [comment]
> The correct way to use the --profile flag with the CLI

As of today, this way didn't work. But setting `AWS_PROFILE` env variable works like a charm!

```
$ tf-next --version
milliVolt CLI 1.0.0-canary.4

1.0.0-canary.4
```

**Comment by ofhouse** [comment]
The original issue has now been fixed in [`v1.0.0-canary.5`](https://github.com/milliHQ/terraform-aws-next-js/releases/tag/v1.0.0-canary.5). Thanks for reporting!

**Comment by Acsigen** [comment]
> @Acsigen Difficult to say, never worked with a roleAssumer for a profile before. Could you provide a reproduction (what I need to create in IAM) to test this?

I use a role that has full access (admin) to that account, the policy looks like this:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": "*",
            "Resource": "*"
        }
    ]
}
```
